src/components/ShareStructure/ShareStructure.vue

```diff
@@ -791,6 +815,11 @@ export default class ShareStructure extends Mixins(CommonMixin) {
     }
   }
 
+  /** Map an empty 2d array to handle the unknown size of nested Share Classes and Series */
+  private mapEmpty2dArray (): Array<any> {
+    return new Array(50).fill(null).map(() => new Array(50).fill(null))
```

Comment:
Not sure if you need this. I believe, in JS, arrays auto-resize so you can just ref an item (and don't have to pre-size the array).

Reply:
https://stackoverflow.com/a/33903611/8679162

Reply:
Or does this not work for 2D arrays?

Reply:
I did have issues managing this yea, specifically because it was a multidimensional array.
In js, i don't think multidimensional arrays really exist natively or are supported, it just boils down to arrays within an array. 
The issue was setting a baseline from which to work with / reset too. 
I can dig into it and see if there is another option tho!

Reply:
Nah, it's OK for now.

Change it later if this keeps you up at night.